### ISSUE TITLE
Help: "Error while loading shared binaries" -- "Invalid ELF Header" after Windows Defender threat 

### ISSUE THREAD

**Issue by Dakedres** [microsoft/WSL#4745]
I was simply running `ranger` on my Debian WSL setup, when Windows Defender alterted me of what it thought was a Trojan

![image](https://user-images.githubusercontent.com/36781227/70383323-429a8d80-1929-11ea-87a4-3c77897b4fcc.png)

From the fact that the file's in the WSL program directory, and it happened right after I ran a command, I assumed it could be a false alarm, but I couldn't be too sure so I simply quarantined it.

However, now every time I run apt, I get this error:
```apt-get: error while loading shared libraries: /lib/x86_64-linux-gnu/libbz2.so.1.0: invalid ELF header```

 I do assume that the ELF header issue is because of Windows Defender, but I'm not sure whether this is actually a threat or just a false alarm or not, some clarification and/or investigation would be appreciated.

**Comment by albuch** [comment]
I have the same issue and get the error message when trying to run any `apt` commands in WSL Ubuntu.
```
apt-get: error while loading shared libraries: /usr/lib/x86_64-linux-gnu/libapt-private.so.0.0: invalid ELF header
``` 
Restoring the file from Windows defender doesn't fix the issue. Basically Windows Defender screwed up whole WSL.

**Comment by benhillis** [comment]
I received word from the Defender team that this has been fixed in security intelligence version 1.307.77.0 and up on the 8th December.

**Comment by jesigloot** [comment]
Took the time to fix my shell now, so for anyone else finding this:
```
wget https://launchpad.net/ubuntu/+source/apt/1.6.11/+build/16829590/+files/apt_1.6.11_amd64.deb
sudo dpkg -i apt_1.6.11_amd64.deb 
```
I can now do apt commands again 🎉

**Comment by Bubesh958** [comment]
When I try to run 
`dpkg -i apt_1.6.11_amd64.deb` (as root,wsl as administrator)
It gives permission denied error
`Preparing to unpack apt_1.6.11_amd64.deb ...
Unpacking apt (1.6.11) over (1.6.11) ...
dpkg: error processing archive apt_1.6.11_amd64.deb (--install):
 unable to make backup link of './usr/lib/x86_64-linux-gnu/libapt-private.so.0.0.0' before installing new version: Permission denied
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Processing triggers for libc-bin (2.27-3ubuntu1) ...
/sbin/ldconfig.real: /usr/lib/x86_64-linux-gnu/libapt-private.so.0.0.0 is not an ELF file - it has the wrong magic bytes at the start.
Errors were encountered while processing:
 apt_1.6.11_amd64.deb`

**Comment by jesigloot** [comment]
@Bubesh958 I might have also removed the file from quarantine and some other things like update/restart. Not sure if any of those helped me as well? :)

**Comment by Dakedres** [comment]
I thought this issue had been left dormant, then I realised I just wasn't recieving any notifications for it haha

Thanks for clarifying that it was not infact an attack, and providing a command to fix apt!

**Comment by Bubesh958** [comment]
@jesigloot  Yeah, removing the file from quarantine and restarting the system helped. Thank you.

**Comment by ikay13** [comment]
When I try
`sudo dpkg -i apt_1.6.11_amd64.deb `
I gives an error processing archive message:
`dpkg: error processing archive apt_1.6.11_amd64.deb (--install):`
` package architecture (amd64) does not match system (armhf)`
`Errors were encountered while processing:`
` apt_1.6.11_amd64.deb`

Any recommendations?

**Comment by ldemattos** [comment]
Hello, fellas!

I was able to recover my apt after installing with dpkg those two debs:
`http://launchpadlibrarian.net/440399778/libapt-pkg5.0_1.6.12_amd64.deb`
`http://launchpadlibrarian.net/440399776/apt_1.6.12_amd64.deb`

After that the coomand `sudo apt --fix-broken install` corrected the loosen tips.

Cya!

**Comment by Jonathannsegal** [comment]
I found that Apt was not able to be installed because of missing the `libapt-pkg5.0_1.6.12ubuntu0.1_amd64.deb` this walkthough was a big help: https://askubuntu.com/questions/178879/apt-get-broken-cannot-fix-or-remove-libapt-pkg4-12

**Comment by therealkenc** [comment]
/fixed "security intelligence version 1.307.77.0" or better.
